retriever/lib/engine_tools.py

```diff
@@ -101,37 +104,102 @@ def reset_retriever(scope="all", ask_permission=True):
             print("can't find script {scp}".format(scp=scope))
 
 
-def json2csv(input_file, output_file=None, header_values=None, encoding=ENCODING):
-    """Convert Json file to CSV.
-
-    Function is used for only testing and can handle the file of the size.
-    """
+def json2csv(input_file,
+             output_file=None,
+             header_values=None,
+             encoding=ENCODING,
+             row_key=None):
+    """Convert Json file to CSV."""
     file_out = open_fr(input_file, encoding=encoding)
     # set output file name and write header
     if output_file is None:
         output_file = os.path.splitext(os.path.basename(input_file))[0] + ".csv"
     csv_out = open_fw(output_file, encoding=encoding)
     if os.name == 'nt':
-        outfile = csv.DictWriter(csv_out,
-                                 dialect='excel',
-                                 escapechar="\\",
-                                 lineterminator='\n',
-                                 fieldnames=header_values)
+        outfile = csv.writer(csv_out,
+                             dialect='excel',
+                             escapechar="\\",
+                             lineterminator='\n')
     else:
-        outfile = csv.DictWriter(csv_out,
-                                 dialect='excel',
-                                 escapechar="\\",
-                                 fieldnames=header_values)
-    raw_data = json.loads(file_out.read())
-    outfile.writeheader()
+        outfile = csv.writer(csv_out, dialect='excel', escapechar="\\")
 
-    for item in raw_data:
-        outfile.writerow(item)
+    raw_data = json.loads(file_out.read(), object_pairs_hook=OrderedDict)
+
+    raw_data, header_values = walker(raw_data,
+                                     row_key=row_key,
+                                     header_values=header_values,
+                                     rows=[],
+                                     normalize=False)
+
+    if isinstance(raw_data[0], dict):
+        raw_data = [list(row.values()) for row in raw_data]
+    else:
+        raw_data = [row.tolist() for row in raw_data]
+    if header_values:
+        outfile.writerow(header_values)
+    outfile.writerows(raw_data)
     file_out.close()
     subprocess.call(['rm', '-r', input_file])
     return output_file
 
 
+def walker(dictionary, row_key=None, header_values=None, rows=[], normalize=False):
+    """
+    Extract rows of data from json datasets
+    """
+    #  Handles the simple case, where row_key and column_key are not required
+    if not (row_key or header_values):
+        if isinstance(dictionary, dict):
+            rows = pd.DataFrame([dictionary]).values
+            header_values = dictionary.keys()
+            return rows, header_values
+        elif isinstance(dictionary, list):
+            rows = pd.DataFrame(dictionary, columns=header_values).values
+            header_values = [chr(i) for i in range(ord('a'), ord('a') + len(rows[0]))]
```

Comment:
I don't understand the code here.

Reply:
That part of code is to create Header values if they don't exist with alphabets.

Reply:
something like `a , b, c, d` right? we should add a comment here mentioning that.
wondering if
```
from string import ascii_lowercase
num_columns = len(rows[0])
header_values = list(ascii_lowercase)[:num_columns]
```
would give a similar result.

Reply:
Added a comment for it

Reply:
@apoorvaeternity ascii_lowercase  works only for 26 columns and it will show error after that. Could you help me in making it work?
`header_values = ['a', 'b', 'c', 'd', 'e', 'f', 'g', 'h', 'i', 'j', 'k', 'l', 'm', 'n', 'o', 'p', 'q', 'r', 's', 't', 'u', 'v', 'w', 'x', 'y', 'z']`
What we want 
`header_values = ['a', 'b', 'c', 'd', 'e', 'f', 'g', 'h', 'i', 'j', 'k', 'l', 'm', 'n', 'o', 'p', 'q', 'r', 's', 't', 'u', 'v', 'w', 'x', 'y', 'z', 'aa', 'ab', 'ac', .....]`

Reply:
Well in that case you can create a generator using `iterator.product()`. Something like this should do it: https://stackoverflow.com/a/42176541

Reply:
@apoorvaeternity Thanks for the suggestion, fixed that issue now